waiter/src/waiter/service_description.clj

```diff
@@ -1252,22 +1253,24 @@
   [reference-type->stale-fn references]
   (every? (fn [reference] (stale-reference? reference-type->stale-fn reference)) references))
 
-(defn service-id->idle-timeout
-  "Computes the idle timeout, in minutes, for a given service.
-   If the service is active or was created by on-the-fly, the idle timeout is retrieved from the service description.
-   Else, the idle timeout is the sum of the fallback period seconds and the stale service timeout."
+(defn service->gc-time
+  "Computes the time when a given service should be GC-ed.
+   If the service has GC disabled (by configuring idle-timeout-mins=0), nil is returned.
+   If the service is active or was created by on-the-fly, the GC time is retrieved from the service description.
```

Comment:
```suggestion
   If the service is active or was created by on-the-fly, the GC time is calculated using
   the service's idle-timeout-mins and the time of the service's most recent request.
```

Reply:
I still think this comment needs rewording, but based on the clarification I just got from you about why we're using last-modified-time, my suggestion is obviously not accurate.

Reply:
The updated docstring uses: `the most recent completion time of the service's requests`